lib/debug/src/transform.rs

```diff
@@ -0,0 +1,581 @@
+pub use crate::read_debuginfo::DebugInfoData;
+use cranelift_codegen::isa::TargetFrontendConfig;
+use cranelift_entity::EntityRef;
+use std::collections::HashMap;
+use wasmtime_environ::AddressTransforms;
+
+use gimli;
+
+use gimli::{
+    AttributeValue, CompilationUnitHeader, DebugAbbrev, DebugAddr, DebugAddrBase, DebugLine,
+    DebugStr, DebuggingInformationEntry, LocationLists, RangeLists, UnitOffset,
+};
+
+use gimli::write;
+
+trait Reader: gimli::Reader<Offset = usize> {}
+
+impl<'input, Endian> Reader for gimli::EndianSlice<'input, Endian> where Endian: gimli::Endianity {}
+
+#[derive(Debug)]
+pub enum Error {
```

Comment:
At this point I use `failure::Error` for 99% of my errors -- it might work for y'all as well.

Reply:
it is useful, though I could not make it work for this case yet. I'll try to add failure dependency in the future.